ethcore/src/state/mod.rs

```diff
@@ -186,8 +194,8 @@ impl AccountEntry {
 /// checkpoint can be discateded with `discard_checkpoint`. All of the orignal
 /// backed-up values are moved into a parent checkpoint (if any).
 ///
-pub struct State {
-	db: StateDB,
+pub struct State<B> {
```

Comment:
I'd agree if there was some generic functionality, but `State` is pretty much useless without a backend. Attempting to use it with unsupported type will leave you with a bare struct which does not really represent the `State` abstraction. I think it would be better to explicitly disallow that.